test/conformance/service_test.go

```diff
@@ -303,114 +304,38 @@ func TestUpdateRevisionTemplateSpecMetadata(t *testing.T) {
-	releaseImagePath3 := test.ImagePath(helloworld)
-
-	names := test.ResourceNames{
-		Service: test.AppendRandomString("pizzaplanet-service", logger),
-	}
-
-	defer tearDown(clients, names)
-	test.CleanupOnInterrupt(func() { tearDown(clients, names) }, logger)
-
-	logger.Info("Creating a new Service in runLatest")
-	svc, err := test.CreateLatestService(logger, clients, names, releaseImagePath1)
-	if err != nil {
-		t.Fatalf("Failed to create Service: %v", err)
-	}
-	names.Route = serviceresourcenames.Route(svc)
-	names.Config = serviceresourcenames.Configuration(svc)
-
-	logger.Info("The Service will be updated with the name of the Revision once it is created")
-	revisionName, err := waitForServiceLatestCreatedRevision(clients, names)
-	if err != nil {
-		t.Fatalf("Service %s was not updated with the new revision: %v", names.Service, err)
-	}
-	names.Revision = revisionName
-	firstRevision := revisionName
-
-	logger.Info("The Service will be updated with the domain of the Route once it is created")
-	routeDomain, err := waitForServiceDomain(clients, names)
-	if err != nil {
-		t.Fatalf("Service %s was not updated with the new route: %v", names.Service, err)
-	}
-
-	logger.Info("When the Service reports as Ready, everything should be ready")
-	if err := test.WaitForServiceState(clients.ServingClient, names.Service, test.IsServiceReady, "ServiceIsReady"); err != nil {
-		t.Fatalf("The Service %s was not marked as Ready to serve traffic to Revision %s: %v", names.Service, names.Revision, err)
-	}
-	assertServiceResourcesUpdated(t, logger, clients, names, routeDomain, "1", "What a spaceport!")
-
-	// Everything above here is setup to get us into a good state to test release mode
-	logger.Info("Updating Service to ReleaseType using lastCreatedRevision")
-	svc, err = test.UpdateReleaseService(logger, clients, svc, []string{firstRevision}, 0)
-	if err != nil {
-		t.Fatalf("Service %s was not updated to release: %v", names.Service, err)
-	}
-
-	logger.Info("Service traffic should go to the first revision and be available on two names traffic targets, 'current' and 'latest'")
-	validateDomains(t, logger, clients,
-		routeDomain,
-		[]string{expectedBlue},
-		[]string{"latest", "current"},
-		[]string{expectedBlue, expectedBlue})
+func updateServiceWithAnnotations(clients *test.Clients, names test.ResourceNames, annotations map[string]string) error {
+	return patchService(
+		clients,
+		names,
+		"add",
+		"/spec/runLatest/configuration/revisionTemplate/metadata/annotations",
+		annotations)
+}
 
-	logger.Info("Updating the Service Spec with a new image")
-	if _, err := test.UpdateServiceImage(clients, svc, releaseImagePath2); err != nil {
-		t.Fatalf("Patch update for Service %s with new image %s failed: %v", names.Service, releaseImagePath2, err)
+func patchService(clients *test.Clients, names test.ResourceNames, operation string, path string, value interface{}) error {
+	patches := []jsonpatch.JsonPatchOperation{
+		{
```

Comment:
nit: `{{` on previous line reduces indent.

I think @dgerd also added a `createPatch` call taking before/after objects, which might make it easier to do this without a deep knowledge of JSON Patch?